lib/puppet/application/resource.rb

```diff
@@ -138,14 +138,16 @@ def main
       resources = find_or_save_resources(type, name, params)
 
       if options[:to_yaml]
-        text = resources.
-          map { |resource| resource.prune_parameters(:parameters_to_include => @extra_params).to_hierayaml }.
-          join("\n")
+        text = resources.map do |resource|
+          r = resource.prune_parameters(:parameters_to_include => @extra_params).to_hierayaml
+          r.encoding == Encoding.default_external? ? r : r.force_encoding(Encoding.default_external).scrub!
+        end.join("\n")
         text.prepend("#{type.downcase}:\n")
       else
-        text = resources.
-          map { |resource| resource.prune_parameters(:parameters_to_include => @extra_params).to_manifest }.
-          join("\n")
+        text = resources.map do |resource|
+          r = resource.prune_parameters(:parameters_to_include => @extra_params).to_manifest
+          r.encoding == Encoding.default_external ? r : r.force_encoding(Encoding.default_external).scrub!
```

Comment:
Can't use `scrub!` as it's not in ruby 1.9.3

Appears we can do this to replicate, even if the string begins in UTF-8
```
'string'.force_encoding(Encoding::UTF_8, Encoding::BINARY, :undef => :replace, :invalid => :replace)
```